espflash/src/bin/espflash.rs

```diff
@@ -235,6 +236,16 @@ fn flash(args: FlashArgs, config: &Config) -> Result<()> {
     // Read the ELF data from the build path and load it to the target.
     let elf_data = fs::read(&args.image).into_diagnostic()?;
 
+    if args.flash_args.image.check_app_descriptor.unwrap_or(true) {
```

Comment:
We also need this check in cargo-espflash. I suggest taking the guts of this block and extracting into a function in the library, which both binaries can call to do the check.